sdk/appconfiguration/azure-data-appconfiguration/src/test/java/com/azure/data/appconfiguration/ConfigurationClientTestBase.java

```diff
@@ -488,4 +510,64 @@ static void assertRestException(Throwable exception, Class<? extends HttpRespons
             assertEquals(exception, ex.getClass());
         }
     }
+
+    /**
+     * Helper method to verify that two configuration setting are equal. Users can defined their equal method.
+     *
+     * @param o1 ConfigurationSetting object 1
+     * @param o2 ConfigurationSetting object 2
+     * @return boolean value that defines if two ConfigurationSettings are equal
+     */
+    boolean equals(ConfigurationSetting o1, ConfigurationSetting o2) {
+        if (o1 == o2) {
+            return true;
+        }
+
+        if (!Objects.equals(o1.getKey(), o2.getKey())
+            || !Objects.equals(o1.getLabel(), o2.getLabel())
+            || !Objects.equals(o1.getValue(), o2.getValue())
+            || !Objects.equals(o1.getETag(), o2.getETag())
+            || !Objects.equals(o1.getLastModified(), o2.getLastModified())
+            || !Objects.equals(o1.isLocked(), o2.isLocked())
+            || !Objects.equals(o1.getContentType(), o2.getContentType())
+            || ImplUtils.isNullOrEmpty(o1.getTags()) != ImplUtils.isNullOrEmpty(o2.getTags())) {
+            return false;
+        }
+
+        if (!ImplUtils.isNullOrEmpty(o1.getTags())) {
+            return Objects.equals(o1.getTags(), o2.getTags());
```

Comment:
Is this going to perform an object reference comparison rather than the contents?

Reply:
getTags() returns Map.  equals(Map, Map) compares contents as well. https://howtodoinjava.com/java/collections/hashmap/compare-two-hashmaps/

Reply:
Hmm interesting. Could you simplify this logic to:
```java
return Objects.equals(o1.getKey(), o2.getKey())
    && Objects.equals(o1.getLabel(), o2.getLabel())
    && Objects.equals(o1.getValue(), o2.getValue())
    && Objects.equals(o1.getETag(), o2.getETag())
    && Objects.equals(o1.getLastModified(), o2.getLastModified())
    && Objects.equals(o1.isLocked(), o2.isLocked())
    && Objects.equals(o1.getContentType(), o2.getContentType())
    && ImplUtils.isNullOrEmpty(o1.getTags()).equals(ImplUtils.isNullOrEmpty(o2.getTags())))
    && Objects.equals(o1.getTags(), o2.getTags());
```